aidial_adapter_openai/utils/prompt_truncate.py

```diff
@@ -0,0 +1,65 @@
+from typing import Callable, List, Set, Tuple, TypeVar
+
+from aidial_sdk.exceptions import (
+    TruncatePromptSystemAndLastUserError,
+    TruncatePromptSystemError,
+)
+
+_T = TypeVar("_T")
+
+DiscardedMessages = List[int]
+UsedTokens = int
+
+
+def truncate_prompt(
+    message_holders: List[_T],
+    message_tokens_getter: Callable[[_T], int],
+    is_system_message_getter: Callable[[_T], bool],
```

Comment:
Why `_getter` suffix? Usually predicates like this one are called `is_SOMETHING`: `is_system_message`. Now it's clear from the name that it's a method, not an object.

Likewise `message_tokens_getter` - if you call a variable using a verb you kill two birds with one stone: (1) it becomes clear it's a method (2) you don't have to spend character on the getter suffix: `tokenize_message`.

Reply:
Because this way user actually understand, that we pass not the value itself, but function, that would get us the value

```python
test_func(is_system_message=test_value) 
```
Here I can easily assume, that we pass boolean values

```python
test_func(is_system_message_getter=test_value)
```
Here I can explicitly see, that this argument is not primitive type, but a getter

Reply:
The types help to resolve the ambiguity in case of `is_system_message`.

What about `tokenize_message`? I don't see any ambiguity here.

Reply:
Ok, let the types do explaining

Removed `_getter` postfix